### PR TITLE
Add support of the time module in Starlark Processor

### LINKED ISSUE #9000: Add support of the time module in Starlark Processor
## Feature Request

The upstream project starlark-go has added [a new module called time](https://github.com/google/starlark-go/pull/327) to support dates and durations, thus the goal of this feature request is to add the support of this new module in the Starlark Processor.  

### Desired behavior:

* May want to consider helpers to parse from Telegraf nanosecond int64 to time and back if it's not obvious
* May want to have examples of string parsing and date string formatting
* a demo plugin where we discard metrics older than 2 weeks might be a good demo of its use.

### PR DESCRIPTION
**Pull request by essobedo**
### Required for all PRs:

- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

fixes #9000

## Motivation

The upstream project starlark-go has added [a new module called time](https://github.com/google/starlark-go/pull/327) to support dates and durations, thus the goal of this feature request is to add the support of this new module in the Starlark Processor.  

## Modifications

* Upgrade the version of `go.starlark.net`
* Import the module `time` of `go.starlark.net` in the Starlark Processor
* Add a test showing how to parse date, duration and timestamp
* Refer the tests into the `README.md`

## Result

In Starlark script, we can now import the module with `load('time.star', 'time')`, and then have access to all the corresponding building functions and constants.

**NB:** Beware the built-in function `from_timestamp` only supports timestamps in seconds.

### CODE REVIEW

**Comment by srebhan** [comment]
Hey @essobedo, thanks for all your effort regarding starlark! You are my hero!
Regarding this PR I think it tries to achieve the same as #7742 and also suffers from the same problems regarding the update of protobuf. How should we proceed? Should I close #7742 in favor of this PR?

**Comment by essobedo** [comment]
@srebhan Hi, if you don't mind closing #7742 in favor of this PR, it would be great

**Comment by essobedo** [comment]
@srebhan about the build failure on `make tidy`, I don't really get it, it doesn't fail on my side. Any idea how to fix it?

**Comment by srebhan** [comment]
Ok closed. Please note, that your build errors/dependency hell is resolved with  #8937 being merged (tested locally).

**Comment by sspaink** [comment]
@essobedo are you using the Go version 1.16+ locally when running `make tidy'? If not could you update Go to the latest version and run it again and push the changes.

**Comment by essobedo** [comment]
> @essobedo are you using the Go version 1.16+ locally when running `make tidy'? If not could you update Go to the latest version and run it again and push the changes.

@sspaink No, I'm using 1.15.8, let me upgrade it and try again, thank you for your help

**Comment by essobedo** [comment]
@sspaink it seems to work now, many thanks

**Comment by sspaink** [comment]
The failing of the job 'share-artifacts' is unrelated to this PR, will be fixed soon but will require a re-base to get the new config. Sorry about that.

**Review by ssoroka — APPROVED** [review]

File: plugins/processors/starlark/testdata/time_timestamp.star [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+load('time.star', 'time')
+# loads time.parse_duration(), time.is_valid_timezone(), time.now(), time.time(), 
+# time.parse_time() and time.from_timestamp()
+
+def apply(metric):
+    # 1616198400 sec = Saturday, March 20, 2021 0:00:00 GMT
+    refDate = time.from_timestamp(1616198400)
+    # 1616020365 sec = Wednesday, March 17, 2021 22:32:45 GMT
+    # 1616150517 sec = Friday, March 19, 2021 10:41:57 GMT
+    metric_date = time.from_timestamp(int(metric.time / 1000000000))
```
Comment (ssoroka) [inline plugins/processors/starlark/testdata/time_timestamp.star:20]: let's change this example to use the from_timestamp nanoseconds as well

**Review by ssoroka — APPROVED** [review]

**Comment by sjwang90** [comment]
@essobedo Do you think you could add `time.star` and `math.star` information to the [Starlark readme libraries section](https://github.com/influxdata/telegraf/blob/master/plugins/processors/starlark/README.md#libraries-available)?

**Comment by essobedo** [comment]
@sjwang90 yes of course, I will propose a PR asap

**Pull request merged**